x-pack/test/apm_api_integration/common/config.ts

```diff
@@ -47,24 +68,45 @@ export function createTestConfig(config: Config) {
     const services = xPackAPITestsConfig.get('services') as InheritedServices;
     const servers = xPackAPITestsConfig.get('servers');
 
-    const supertestAsApmReadUser = supertestAsApmUser(servers.kibana, ApmUser.apmReadUser);
-
     registry.init(config.name);
 
     return {
       testFiles: [require.resolve('../tests')],
       servers,
       services: {
         ...services,
-        supertest: supertestAsApmReadUser,
-        supertestAsApmReadUser,
-        supertestAsNoAccessUser: supertestAsApmUser(servers.kibana, ApmUser.noAccessUser),
-        supertestAsApmWriteUser: supertestAsApmUser(servers.kibana, ApmUser.apmWriteUser),
-        supertestAsApmAnnotationsWriteUser: supertestAsApmUser(
+
+        supertestClients: async (context: InheritedFtrProviderContext) => {
```

Comment:
when is this function called? on every test run, or for every individual test?

Reply:
and do you think we can maybe drop the `supertest` part and name it something more appropriately? e.g. `apmApiClient`?

Reply:
Renamed to `apmApiClients`

Reply:
> when is this function called? on every test run, or for every individual test?

Will be called when the service is loaded:

```
const apmApiClient = getService('apmApiClient');
```